internal/server/static_fs.go

```diff
@@ -35,8 +35,13 @@ const uiFilePathPrefix = "ui"
 
 func (sfs StaticFileSystem) Exists(prefix string, filepath string) bool {
 	if p := strings.TrimPrefix(filepath, prefix); len(p) < len(filepath) {
-		_, err := sfs.base.Open(path.Join(uiFilePathPrefix, p))
-		return err == nil
+		if _, err := sfs.base.Open(path.Join(uiFilePathPrefix, p)); err == nil {
+			return true
+		}
+
+		if _, err := sfs.base.Open(path.Join(uiFilePathPrefix+".html", p)); err == nil {
+			return true
+		}
```

Comment:
Hmm, I thought `TestServer_GenerateRoutes_UI/page without .html suffix` was already testing this case. 

I guess that test is wrong, because `/providers` also exists as a directory. Oops!

Changing the path in that test case to `/providers/add/admins` shows the problem i think. That change should be enough to verify this fix.